website/docs/r/solution_add_on.html.markdown

```diff
@@ -0,0 +1,133 @@
+  Director extensions together with their backend services and lifecycle management. Solution
+  Add-Ons are distributed as .iso files. A Solution Add-on can contain numerous
+  elements: UI plugins, vApps, users, roles, runtime defined entities, and more.
+---
+
+# vcd\_solution\_add\_on
+
+Supported in provider *v3.13+* and VCD 10.4.1+.
+
+Provides a resource to manage Solution Add-Ons in Cloud Director. A Solution Add-On is the
+representation of a solution that is custom built for Cloud Director in the Cloud
+Director extensibility ecosystem. A Solution Add-On can encapsulate UI and API Cloud Director
+extensions together with their backend services and lifecycle management. Solution аdd-оns are
+distributed as .iso files. A Solution Add-On can contain numerous elements: UI plugins, vApps,
+users, roles, runtime defined entities, and more.
+
+~> Only `System Administrator` can create this resource.
+
+## Example Usage (Uploading an image and create a Solution Add-On entry)
+
+```hcl
+data "vcd_solution_landing_zone" "slz" {}
+
+resource "vcd_catalog_media" "dse14" {
+  org        = "solution_org"
+  catalog_id = tolist(data.vcd_solution_landing_zone.slz.catalog)[0].id
+
+  name              = basename("/Users/demo/Downloads/vmware-vcd-ds-1.4.0-23376809.iso")
+  description       = "new os versions"
+  media_path        = "/Users/demo/Downloads/vmware-vcd-ds-1.4.0-23376809.iso"
+  upload_any_file   = false # Add-ons are packaged in '.iso' files
+  upload_piece_size = 10
+}
+
+resource "vcd_solution_add_on" "dse14" {
+  catalog_item_id   = data.vcd_catalog_media.dse14.catalog_item_id
+  addon_path        = "/Users/demo/Downloads/vmware-vcd-ds-1.4.0-23376809.iso"
+  trust_certificate = true
+  accept_eula       = true
+}
+
+```
+
+## Example usage (Using already uploaded image)
+```hcl
+data "vcd_solution_landing_zone" "slz" {}
+
+data "vcd_catalog_media" "dse14" {
+  org        = "solution_org"
+  catalog_id = tolist(data.vcd_solution_landing_zone.slz.catalog)[0].id
+
+  name = basename("/Users/demo/Downloads/vmware-vcd-ds-1.4.0-23376809.iso")
+}
+
+resource "vcd_solution_add_on" "dse14" {
+  org               = "solution_org"
+  catalog_item_id   = data.vcd_catalog_media.dse14.catalog_item_id
+  addon_path        = "/Users/demo/Downloads/vmware-vcd-ds-1.4.0-23376809.iso"
+  trust_certificate = true
+  accept_eula       = true
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `catalog_item_id` - (Required) The catalog item ID of Solution Add-On. It can be uploaded using
+  [`vcd_catalog_media`](/providers/vmware/vcd/latest/docs/resources/catalog_media)
+* `addon_path` - (Required) Local filesystem path of Solution Add-On. The Add-On image is required
+locally as some of the metadata must be extracted for creating the Solution Add-On entry in Cloud
+Director.
+* `trust_certificate` - (Optional) Plugin can automatically trust the certificate of Solution
```

Comment:
This property name can be perceived as either `verb+noun` –requires a Boolean– or as `(possessive)noun+noun` –requires a string–. The intention is the first one, but a few times, looking at it, I see it as noun+noun.
Could we do one of the following?
a. find a clearer name
b. improve the documentation to make clear that we expect a Boolean value

Reply:
All of the above look good to me.

Reply:
+1, hard to decide. I'm between `trust_certificate_on_create` and `auto_trust_certificate`:
* First one states that is a verb + clarifies when it applies
* Second one corresponds with the description

Reply:
Current one is fine with a doc update:

>  b. improve the documentation to make clear that we expect a Boolean value

`trust_certificate_on_create` is clear too. Is it only on create, though?

`auto_trust_certificate` - fine too.

Reply:
I think we are converging to `auto_trust_certificate` then. I will double check field, docs and code.

Reply:
Set field name to `auto_trust_certificate`. It works at create phase